fomantic/Fomantic-UI
# fix(dropdown): keep original values, also on remove and more fixes

## Description
Fixed the following issues:
- preset values inside an already existing hidden dropdown input field containing characters like < > " ' were encoded during setup on non-select dropdowns and thus added twice inside the hidden input
- removing selected multiple entries encoded the whole selected string on non-select dropdowns
- entering the same value in multiple search dropdowns as already provided as "values"  on initialize time were ignored/not removed on enter/delimiter
- pressing enter on such already existing values forced a form submit
- placeholder text was encoded twice when preserveHTML was set to false
- when every item was filtered and reopened again via backspace deletion some css was missing for the menu
- hiding menu when all items are filtered left the visible class which prevents another dropdown to be shown with higher z-index
- reshow menu when all non selected items were filtered and an item was selected

## Testcase
- Watch console and see initial form input field
- remove one field and watch the hidden input
- enter "a" and press comma
- enter "a" and press enter

### Broken
https://jsfiddle.net/lubber/61u0xpL8/13/
- the username email is encoded and added to the existing value
- removing one entry will entitiy encode all other values
- pressing comma does nothing, it does not even remove the value 
- pressing enter triggers a form submit

### Fixed
https://jsfiddle.net/lubber/61u0xpL8/20/
- the username email is kept original
- after removing one item the hidden input still contains the untouched values as before
- pressing comma removes the search entry as the value is already selected
- pressing enter also removes the search entry and does not trigger a form submit

## Closes
#2246
